styx-flyte-client/src/main/java/com/spotify/styx/flyte/client/FlyteInputsUtils.java

```diff
@@ -46,7 +40,7 @@ private FlyteInputsUtils() {
   static Literals.Literal buildLiteralForPartition(String value) {
 
     var primitive =
-        Literals.Primitive.newBuilder().setDatetime(toTimestamp(value)).build();
+        Literals.Primitive.newBuilder().setDatetime(parseBest(value)).build();
```

Comment:
Can we get the `schedule` here from workflow definition? If yes we don't need to do this kind of best matching parsing which is both time consuming and error prone.